src/hatti/map/utils.cljs

```diff
@@ -506,6 +506,87 @@
     :else {:layer-type "circle" :style :point}))
 
+(defn to-fixed
+  [n number-of-decimal-places]
+  (when n
+    (-> n
+        (.toFixed number-of-decimal-places)
+        js/parseFloat)))
+
+(defn percent
+  [n total & {:keys [number-of-decimal-places]}]
+  (let [percentage (when (and (pos? n) (>= total n))
+                     (-> n
+                         (/ total)
+                         (* 100)
+                         float))]
+    (if number-of-decimal-places
+      (to-fixed percentage number-of-decimal-places)
+      percentage)))
+
+(defn generate-hexgrid
+  "Generates hexbins with point count aggregation given rendered
+  layer-id or geojson"
+  [map layer-id & [geojson]]
+  (let [get-rendered-features #(.queryRenderedFeatures
+                                map (clj->js {:layers [layer-id]}))
+        rendered-features (or geojson
+                              (clj->js
+                               {:type "FeatureCollection"
+                                :features (get-rendered-features)}))
+        ;; Get bounding box for rendered features
+        bbox (.bbox js/turf (clj->js rendered-features))
+        cellWidth 60
+        units "kilometers"
+        ;; generete hexGrid with bounding box
+        hexgrid (.hexGrid js/turf bbox cellWidth units)
+        ;; collect point IDs within each polygon area
+        hex-collection (.collect js/turf hexgrid (clj->js rendered-features)
+                                 "_id" "points")
+        total (-> rendered-features :features count)
+        hexbins (js->clj hex-collection :keywordize-keys true)
+        ;; Count points on hexbins
+        features-w-count (remove nil?
+                                 (for [{{:keys [points]} :properties :as
+                                        feature}
+                                       (:features hexbins)
+                                       :let [point_count
+                                             (percent (count points) total)]]
+                                   (when point_count
+                                     (assoc feature :properties
+                                            {:point_count point_count}))))]
+    ;; return hexbins with updated point_count
+    (assoc hexbins :features features-w-count
+           :properties {:total
+                        total})))
+
+(defn show-hexbins
+  "Renders hexbin layer on map."
+  [map id_string geojson]
+  (let [id "hexgrid"
+        hexgrid (generate-hexgrid map id_string geojson)
+        _ (.log js/console (clj->js hexgrid))
+        total (-> hexgrid :properties :total)]
+    (add-mapboxgl-source map id {:geojson hexgrid})
+    (add-mapboxgl-layer map id
+                        "fill"
+                        :paint {:fill-outline-color
+                                {:property "point_count"
+                                 :stops [[0 "transparent"]
+                                         [100 "#ccc"]]}
```

Comment:
can this, and colors on line 579 and 580 not be controlled by stylesheets? If not these colors should also go to args or descriptive defs